### ISSUE TITLE
transaction: is it expected, a transaction in CN is divided into multi transactions in DN

### ISSUE THREAD

**Issue by DBInsight** [postgres-x2/postgres-x2#288]
for example:
start transaction;
select \* from t1;
select \* from t2;
select \* from t3;
commit;

this is one transaction in CN, but when they executed on DN, they are divided into 3 transactions:
each select is a individual transction;

cause: in do_query,need_tran_block = false,  so pgxc_node_begin does not send 'start transaction' to DN;

**Comment by koichi-szk** [comment]
Not expected so far for simplicity.   Of course, CN transaction is divided
into multiple transaction in DNs, but each DN has only one of such piece of
transaction.

It may be possible to divide one CN transaction in multiple transactions in
DN if we can maintain all the transaction ACID properties.

---

Koichi Suzuki
https://www.linkedin.com/in/koichidbms

2015-09-01 20:42 GMT+09:00 DBInsight notifications@github.com:

**Comment by DBInsight** [comment]
hmm, i want to know if there is risk for data inconsistency, although now, i can not see the problem

**Comment by koichi-szk** [comment]
If CN transaction is divided into multiple transaction in one DN, we need
to handle such DN transactions as subtransaction.    I'm not sure so far if
existing PG infrastructure can be used for this.

---

Koichi Suzuki
https://www.linkedin.com/in/koichidbms

2015-09-02 11:55 GMT+09:00 DBInsight notifications@github.com:

**Comment by DBInsight** [comment]
we can send down 'start transaction' statement to DN in pgxc_node_begin.

 such, the logical in DN is same to CN, statement in CN and DN are both inside one transaction

**Comment by koichi-szk** [comment]
Right.   Logically, transaction segment in CD and DNs are part of the same
transaction.

---

Koichi Suzuki
https://www.linkedin.com/in/koichidbms

2015-09-02 12:12 GMT+09:00 DBInsight notifications@github.com:
